content/guest-post/node.js-user-authentication-guide/Node_User_Role_auth.md

```diff
@@ -0,0 +1,1224 @@
+
+## Perform CRUD Operations
+
+You'll create functions that handle adding users, getting all users, updating the role of users, and deleting users.
+
+### Register Function
+
+As the name implies, this function will handle the registrations of users.
+
+Let'ss create a new folder named `Auth`. It will contain the Authentication file and the Route set-up file.
+
+After creating the `Auth` folder, add two files to it — `Auth.js` and `Route.js`.
+
+Now open up our `Auth.js` file and import that `User` model:
+
+```js
+const User = require("../model/User")
+```
+
+The next step is to create an `async` `express` function that will take the user's data and register it in the database.
+
+You need to make use of an [Express middleware](https://expressjs.com/en/guide/writing-middleware.html) function that will grant access to the user's data from the body. You'll use this function in the `server.js` file:
+
+```js
+const app = express()
+app.use(express.json())
+```
+
+Let's go back to your `Auth.js` file and create the register funcation:
+
+```js
+// auth.js
+exports.register = async (req, res, next) => {
+  const { username, password } = req.body
+  if (password.length < 6) {
+    return res.status(400).json({ message: "Password less than 6 characters" })
+  }
+  try {
+    await User.create({
+      username,
+      password,
+    }).then(user =>
+      res.status(200).json({
+        message: "User successfully created",
+        user,
+      })
+    )
+  } catch (err) {
+    res.status(401).json({
+      message: "User not successful created",
+      error: error.mesage,
+    })
+  }
+}
+```
+
+The exported `register` function will be used to set up the routes. You got the username and password from the `req.body` and created a `tryCatch` block that will create the user if successful; else, it returns a status code `401` with the error message.
+
+### Set Up Register Route
+
+You'll create a route to `/register` using `express.Router`. Import the `register` function into your `route.js` file, and use it as the route's function:
+
+```js
+const express = require("express")
+const router = express.Router()
+const { register } = require("./auth")
+router.route("/register").post(register)
+module.exports = router
+```
+
+The last step is to import your `route.js` file as middleware in `server.js`:
+
+```js
+app.use("/api/auth", require("./Auth/route"))
+```
+
+The server will use the `router` middleware function if there is a request to `/api/auth`.
+
+### Test the Register Route
+
+You'll use [Postman](https://www.postman.com/downloads/) to test all the routes.
+
+Open up Postman to make a `POST` request to `http://localhost:5000/api/auth/register` and pass the username and password to the body:
+
+![Register Route](https://paper-attachments.dropbox.com/s_1FDB76B1B7350D0381EC73781AAAABB04FC20EA7CD90D74041F3D461F538847D_1639605986714_registerRoute.jpg)
```

Comment:
@uma-victor1 Throughout the blog, please place the image file in the same folder as the blog content and replace the Dropbox link

Reply:
fixed